src/data_types/w3c/credential.rs

```diff
@@ -73,356 +55,120 @@ pub struct CredentialSubject {
         self.proof
-            .get_value(&|proof: &CredentialProof| proof.get_credential_signature_proof())
+            .get_mut_value(&|proof: &mut CredentialProof| match proof {
+                CredentialProof::DataIntegrityProof(proof) => Ok(proof),
+                _ => Err(err_msg!("Credential does not contain data integrity proof")),
+            })
     }
 
-    pub(crate) fn get_mut_credential_signature_proof(
-        &mut self,
-    ) -> Result<&mut CredentialSignatureProof> {
-        self.proof.get_mut_value(&|proof: &mut CredentialProof| {
-            proof.get_mut_credential_signature_proof()
-        })
-    }
+    pub(crate) fn validate(&self) -> Result<()> {
+        let version = self.context.version()?;
 
-    pub fn get_presentation_proof(&self) -> Result<&CredentialPresentationProof> {
-        self.proof
-            .get_value(&|proof: &CredentialProof| proof.get_presentation_proof())
-    }
+        self.context.validate()?;
 
-    pub fn validate(&self) -> Result<()> {
-        if !self.context.0.contains(&URI(W3C_CONTEXT.to_string())) {
-            return Err(err_msg!("Credential does not contain w3c context"));
-        }
-        if !self
-            .context
-            .0
-            .contains(&URI(W3C_ANONCREDS_CONTEXT.to_string()))
-        {
-            return Err(err_msg!(
-                "Credential does not contain w3c anoncreds context"
-            ));
-        }
-        if !self.type_.0.contains(W3C_CREDENTIAL_TYPE) {
+        if !self.type_.0.contains(&W3C_CREDENTIAL_TYPE.to_string()) {
             return Err(err_msg!("Credential does not contain w3c credential type"));
         }
-        if !self.type_.0.contains(W3C_ANONCREDS_CREDENTIAL_TYPE) {
-            return Err(err_msg!(
-                "Credential does not contain w3c anoncreds credential type"
-            ));
-        }
-        Ok(())
-    }
-
-    pub fn get_schema_id(&self) -> &SchemaId {
-        &self.credential_schema.schema
-    }
-
-    pub fn get_cred_def_id(&self) -> &CredentialDefinitionId {
-        &self.credential_schema.definition
-    }
 
-    pub fn get_rev_reg_id(&self) -> Option<&RevocationRegistryDefinitionId> {
-        if let Some(credential_status) = self.credential_status.as_ref() {
-            match credential_status.type_ {
-                CredentialStatusType::AnonCredsCredentialStatusList2023 => {
-                    Some(&credential_status.id)
+        match version {
+            VerifiableCredentialSpecVersion::V1_1 => {
+                if self.issuance_date.is_none() {
+                    return Err(err_msg!(
+                        "V1.1 Credential must include `issuanceDate` property"
+                    ));
+                }
+            }
+            VerifiableCredentialSpecVersion::V2_0 => {
+                if self.valid_from.is_none() {
+                    return Err(err_msg!(
+                        "V1.1 Credential must include `validFrom` property"
```

Comment:
`validFrom` is not required. https://www.w3.org/TR/vc-data-model-2.0/

```suggestion
                        "V2 Credential must include `validFrom` property"
```